bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/protocol/http/TransientCommunicationIssueException.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linktap.protocol.http;
+
+import java.io.Serial;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link TransientCommunicationIssueException} should be thrown when the endpoint being communicated with
+ * does not appear to be a Tap Link Gateway device.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public class TransientCommunicationIssueException extends Exception {
+    @Serial
+    private static final long serialVersionUID = -7786449325604143287L;
+
+    public TransientCommunicationIssueException() {
+        super();
+    }
+
+    public TransientCommunicationIssueException(final String message) {
+        super(message);
+    }
+
+    public TransientCommunicationIssueException(final Throwable cause) {
+        super(cause);
+    }
+
+    public TransientCommunicationIssueException(final String message, final Throwable cause) {
+        super(message, cause);
+    }
+
+    public static final String HOST_UNREACHABLE = "Could not connect";
+    public static final String HOST_NOT_RESOLVED = "Could not resolve IP address";
+    public static final String HOST_COMM_TIMEOUT = "Communications Lost";
```

Comment:
Did you consider to use the i18n provider to support translated error messages?
For reference:https://www.openhab.org/docs/developer/utils/i18n.html#i18n-text-provider-api

Reply:
So I'm working through error messages that will appear on the UI or ones that could appear on the console that could realistically appear in normal operation. However, I am struggling to find examples of parameter values being injected into the i18 value. Do you know of a binding that does this, I can reference or am I missing some documentation somewhere? Sorry - figured I'd ask in case I loose a few days trying to find this :) @lsiepel Thanks in advance - either way.

Reply:
Gree or Salus are probably some nice examples, but there are more. Search for `{}` in *.properties files and you can see what binding use translated strings with arguments

Reply:
See also https://www.openhab.org/javadoc/latest/org/openhab/core/thing/i18n/thingstatusinfoi18nlocalizationservice

Reply:
Thank you @lsiepel and @jlaur :)